pages/Category/components/Empty/selectors.js

```diff
@@ -0,0 +1,34 @@
+import { createSelector } from 'reselect';
+import { isNumber } from '@shopgate/pwa-common/helpers/validation';
+import {
+  getCategoryProductCount,
+  getCurrentCategoryChildCount,
+} from '@shopgate/pwa-common-commerce/category/selectors';
+import { getProductsResult } from '@shopgate/pwa-common-commerce/product/selectors/product';
+
+/**
+ * Determines whether or not a category has any content,
+ * (subcategories, child products or products received via request).
+ */
+export const hasContent = createSelector(
+  getCategoryProductCount,
+  getCurrentCategoryChildCount,
+  getProductsResult,
+  (categoryProductCount, categoryChildrenCount, { products, totalProductCount }) => {
+    /**
+     * Check if we have any products and if we have a valid number
+     * for the total number of products. Not having a valid number
+     * indicates that a request is ongoing.
+     */
+    const hasProductResult = products.length === 0 && isNumber(totalProductCount);
```

Comment:
Hmmm if `products.length === 0` i would say, that no products result is available.

Reply:
Both of these checks are necessary for it to work.